### PR TITLE
Update instructions for stage 2 and make C++ stage 1 diff smaller

### PR DESCRIPTION
**Pull request by andy1li**
image

### CODE REVIEW

File: compiled_starters/cpp/src/main.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@ int main() {
 
   // Uncomment this block to pass the first stage
   // std::cout << "$ ";
-  //
-  // std::string input;
-  // std::getline(std::cin, input);
+  
+  std::string input;
+  std::getline(std::cin, input);
```
Comment (rohitpaulk) [inline compiled_starters/cpp/src/main.cpp:12]: @andy1li this is a good idea, but just want to confirm - will this cause the tester to end up waiting 10 seconds (test timeout) for the program to terminate? Or will we terminate early?

File: solutions/cpp/01-oo8/code/src/main.cpp [hunk 2]
```diff
@@ -6,7 +6,7 @@ int main() {
   std::cerr << std::unitbuf;
 
   std::cout << "$ ";
-
+
```
Comment (rohitpaulk) [inline solutions/cpp/01-oo8/code/src/main.cpp:9]: @andy1li let's remove trailing spaces

**Review by rohitpaulk — APPROVED** [review]

File: compiled_starters/cpp/src/main.cpp (hunk 1 above)
Comment (andy1li) [reply compiled_starters/cpp/src/main.cpp:12]: Don't know yet, but I can test and report back.

For context, all starter code expect cpp are already looking like this. C++ is the odd one out.

Comment (andy1li) [reply compiled_starters/cpp/src/main.cpp:12]: @rohitpaulk  Result: We terminate early when "$ " is not found:

image

File: solutions/cpp/01-oo8/code/src/main.cpp (hunk 2 above)
Comment (andy1li) [reply solutions/cpp/01-oo8/code/src/main.cpp:9]: Look like it's actually for indentation. Without it the diff will behave weirdly.

image

File: compiled_starters/cpp/src/main.cpp (hunk 1 above)
Comment (rohitpaulk) [reply compiled_starters/cpp/src/main.cpp:12]: Gotcha, looks good!

File: solutions/cpp/01-oo8/code/src/main.cpp (hunk 2 above)
Comment (rohitpaulk) [reply solutions/cpp/01-oo8/code/src/main.cpp:9]: @andy1li yep, but for empty lines we don't need indentation (I don't think we have it for the other lines)

Comment (andy1li) [reply solutions/cpp/01-oo8/code/src/main.cpp:9]: Ah, got it! (The weird diff came from the stage 2 solution. Changing that as well.)

File: compiled_starters/cpp/src/main.cpp (hunk 1 above)
Comment (andy1li) [reply compiled_starters/cpp/src/main.cpp:12]: @rohitpaulk Does the () look weird to you as well? I don't remember if it's there before.

image

**Pull request merged**